rmg.py

```diff
@@ -69,94 +70,105 @@ def parseCommandLineArguments(args):
+    parser.add_argument('-p', '--profile', action='store_true',
+                        help='run under cProfile to gather profiling statistics, and postprocess them if job completes')
+    parser.add_argument('-P', '--postprocess', action='store_true',
+                        help='postprocess profiling statistics from previous [failed] run; does not run the simulation')
 
     parser.add_argument('-t', '--walltime', type=str, nargs=1, default='00:00:00:00',
-        metavar='DD:HH:MM:SS', help='set the maximum execution time')
-
-    #Add option to output a folder that stores the details of each kinetic database entry source
-    parser.add_argument('-k', '--kineticsdatastore', action='store_true', help='output a folder, kinetics_database, that contains a .txt file for each reaction family listing the source(s) for each entry')
+                        metavar='DD:HH:MM:SS', help='set the maximum execution time')
 
-    return parser.parse_args(args)
+    # Add option to output a folder that stores the details of each kinetic database entry source
+    parser.add_argument('-k', '--kineticsdatastore', action='store_true',
+                        help='output a folder, kinetics_database, that contains a .txt file for each reaction family '
+                             'listing the source(s) for each entry')
 
-################################################################################
+    args = parser.parse_args(command_line_args)
 
-if __name__ == '__main__':
-
-    # Parse the command-line arguments (requires the argparse module)
-    args = parseCommandLineArguments(sys.argv[1:])
+    # Process args to set correct default values and format
 
     # For output and scratch directories, if they are empty strings, set them
     # to match the input file location
-    import os.path
-    inputFile = args.file[0]
-
-    if args.output_directory != '':
-        args.output_directory = args.output_directory[0]
+    args.file = args.file[0]
 
+    # If walltime was specified, retrieve this string from the element 1 list
     if args.walltime != '00:00:00:00':
         args.walltime = args.walltime[0]
 
-    inputDirectory = os.path.abspath(os.path.dirname(inputFile))
+    # Set directories
+    input_directory = os.path.abspath(os.path.dirname(args.file))
+
     if args.output_directory == '':
-        args.output_directory = inputDirectory
-    if args.scratch_directory == '':
-        args.scratch_directory = inputDirectory
+        args.output_directory = input_directory
+    # If output directory was specified, retrieve this string from the element 1 list
+    else:
+        args.output_directory = args.output_directory[0]
 
     if args.postprocess:
-        print "Postprocessing the profiler statistics (will be appended to RMG.log)"
         args.profile = True
+
+    return args
+
+
+def main():
+    # Parse the command-line arguments (requires the argparse module)
+    args = parse_command_line_arguments(sys.argv[1:])
```

Comment:
I found out that the `parse_args` method defaults to parsing `sys.argv[1:]`, so I think it would be cleaner to have `parse_command_line_arguments()` take an optional `command_line_args` with a default of `None`. That would still allow testing it the way you are currently, and not require an argument be provided when using it here and in generateReactions.py.

Reply:
Thanks for the suggestion! I was unaware that the `parse_args` method worked this way by default. I think this is a great idea. I'll make these changes and rebase.